sys/dev/aic7xxx/aic79xx.c

```diff
@@ -6261,8 +6264,22 @@ ahd_init(struct ahd_softc *ahd)
 	/*
 	 * Only allow target mode features if this unit has them enabled.
 	 */
-	if ((AHD_TMODE_ENABLE & (0x1 << ahd->unit)) == 0)
+#ifdef AHD_TARGET_MODE
+	tmode_enable = ((AHD_TMODE_ENABLE & (0x1 << ahd->unit)) != 0);
+	resource_int_value(device_get_name(ahd->dev_softc),
+			       device_get_unit(ahd->dev_softc),
+			       "tmode_enable", &tmode_enable);
+
+	if (tmode_enable == 0) {
 		ahd->features &= ~AHD_TARGETMODE;
+	} else {
+		if (bootverbose)
+			printf("%s: enabling target mode\n", ahd_name(ahd));
```

Comment:
On a second thought, this message would be confusing if (ahd->features & AHD_TARGETMODE) is false for some other reason, like not supported by hardware/firmware.  Can it be?

Reply:
You are right, it can be for cards that do not support it at all. Good catch, thank you. I will fix that.

Reply:
Fixed now, thank you.